django/thunderstore/frontend/views.py

```diff
@@ -1,23 +1,85 @@
 from django.conf import settings
 from django.shortcuts import render
+from django.utils.decorators import classonlymethod
+from django.views import View
 from django.views.generic import TemplateView
 
+from thunderstore.repository.mixins import CommunityMixin
+
 
 def disable_cache_if_static_request(request, response):
     if request.path.startswith(settings.STATIC_URL):
         response["Cache-Control"] = "no-cache"
 
 
-def handle404(request, exception=None):
-    response = render(request, "errors/404.html", locals(), status=404)
-    disable_cache_if_static_request(request, response)
-    return response
+class ErrorHandler(View, CommunityMixin):
```

Comment:
Why is this required specifically? If it's to insert the appropriate community context, you're probably better off removing the need for said context in the templates

Reply:
In any case I'd go for a solution that avoids us defining this class and I'm fairly sure a reasonable one should exist

Reply:
I could remove the context from the error pages, but then the links would be defaulted to riskofrain2 in the new URLs. Also I'd need to add another base.html file or insert the all the html in the error template files.

I felt like this was the cleanliest way of doing this, since I'm kinda doing the Django update next, which means I can remove the `as_view` method.

Also using class-based views seemed nicer and more inline with the rest of the whole sites views 🤷‍♂️

Let me know what you prefer, since I'm not against anything, just put this together as an draft I guess 👌

Reply:
Why is this necessary only after these changes & not before? If the context was available before, can this logic be inserted there?

Reply:
The `context_data` wasn't available before, it has been using the data provided by the middleware.

Reply:
And I really don't want to mash this community context stuff into the middleware, since we've been trying to get rid of the middleware in some sense.

Reply:
So in general I'd rather have the error views be without context instead of this hacky solution, but I'd also be surprised if there's no less hacky way to get the context included in the error views.

I can check for some suggestions if you want, but right off the bat I could think of writing a function that reads the URL parameters outside of a hacky class based view setup. You might also be able to instantiate the `CommunityMixin` and call `get_context_data` on it without actually inheriting it in case the inheritance is causing the view hackiness.

Reply:
I get that the `as_view` thing is hacky, but it will be removed in the Django/Python update that I'll be working on quite next. IMO this solution is the probably the least hackiest, since it removes the need for separate error templates and doesn't change anything from the users perspective. And keeps the views in same "ideological state".

I'd strongly prefer going with this and not using anymore time on a matter that is guaranteed to solve it self in the near future.